dftimewolf/lib/collectors/osquery.py

```diff
@@ -45,11 +47,95 @@ def _ValidateOsquery(self, query: str) -> bool:
     # TODO(sydp): add more checks.
     return query.upper().startswith('SELECT ')
 
+  def _ParsePlatforms(self, platforms: str) -> List[str]:
+    """Parse and normalise the platforms value from an osquery pack.
+
+    Arguments:
+      platforms: the platforms value from an osquery pack.
+
+    Returns:
+      a list of operating system platforms.
+    """
+    if not platforms:
+      return []
+
+    unique_platforms = set()
+    for platform in platforms.split(','):
+      platform = platform.strip()
+      if platform in ('all', 'any'):
+        unique_platforms.update(_ALL_PLATFORMS)
+      elif platform == 'posix':
+        unique_platforms.update(['darwin', 'freebsd', 'linux'])
+      elif platform in _ALL_PLATFORMS:
+        unique_platforms.add(platform)
+      else:
+        self.logger.warning(f'Unexpected value {platform} in platform value.')
+
+    return list(unique_platforms)
+
+  def _LoadOsqueryPack(self, path: str) -> None:
+    """Loads osquery from an osquery pack file.
+
+    Args:
+      path: the path to the JSON file.
+    """
+    with open(path, mode='r') as fd:
+      global_platform = []
+
+      query_pack = json.load(fd)
+
+      # A 'global' platform value can be set at the root level
+      if 'platform' in query_pack:
+        global_platform = self._ParsePlatforms(query_pack.get('platform'))
+
+      for num, (name, entry) in enumerate(
+          query_pack.get('queries', {}).items()):
+        if not self._ValidateOsquery(entry.get('query', '')):
```

Comment:
Should we just access `entry['query']` and catch a KeyError? I feel it's safe to assume that the dicts will be well-constructed.

Reply:
Also, given that you're using it below, i would just store this in an intermediate variable.

Reply:
Done.